### ISSUE TITLE
Sampler only sample samples

### ISSUE THREAD

**Issue by spider312** [GoXLR-on-Linux/goxlr-utility#195]
Hi, and sorry if i did something wrong, but i can't understand how the sampler works

I can create samples from UI, importing an MP3 and playing it when i push corresponding sample button

But when i try to create samples from my mic, vocal chat or anything else by pressing a red sampler button, it doesn't works, the buttons stays red
And  when i play a sample created with UI, and try to sample it, sampler works as intended, creating a sample of the sample, button becomes green and plays the sample when i push it

Here is my routing, but i tried many settings, none of them allowing me to record anything else than my samples, nor forbidding me to record them
![Routing](https://github.com/user-attachments/assets/e1da9902-6a7f-4c6b-a8e5-6c37921d7f21)

Here is the subsection of my log when i push the button
```
15:30:23 [DEBUG] (2) goxlr_daemon::device: Setting Routing: Samples LineOut true
15:30:23 [DEBUG] (2) goxlr_daemon::device: Applying Routing to Samples:
15:30:23 [DEBUG] (2) goxlr_daemon::device: {Headphones: true, BroadcastMix: true, ChatMic: true, Sampler: false, LineOut: true}
15:30:23 [DEBUG] (2) goxlr_daemon::primary_worker: Refreshing App Path..
15:30:23 [DEBUG] (2) goxlr_daemon::platform: Refreshing App Path..
15:30:24 [DEBUG] (2) goxlr_daemon::device: Handling Button Down: SamplerBottomRight
15:30:24 [DEBUG] (2) goxlr_daemon::device: Handling Sample Button, clear state: false
15:30:24 [DEBUG] (15) goxlr_daemon::device: Handling Button Hold: SamplerBottomRight
15:30:26 [DEBUG] (16) goxlr_daemon::device: Handling Button Release: SamplerBottomRight, Has Long Press Handled: true
15:30:26 [INFO] No Noise Received, or error in recording, Cancelling.
15:30:26 [DEBUG] (16) goxlr_daemon::audio: Calculated Gain: 2
```

**Comment by FrostyCoolSlug** [comment]
The sampler will record so long as you're holding a sample button (the sample light should flash while it's recording).

As for what it records, if you look at the sampler 'output' row on the routing table, it will record from any inputs which are ticked, so currently you're only recording from the microphone, if you add other channels it'll mix and record them as well, so depending on what you plan to sample, you might need to tick a few more columns :)

Let me know if you have any other questions.

**Comment by spider312** [comment]
The problem is my sampler doesn't record the mic with those settings, and it even record sampler when i untick samples on the sample line ...

**Comment by FrostyCoolSlug** [comment]
Could you provide me with the full log file please?

**Comment by spider312** [comment]
[goxlr-daemon.log](https://github.com/user-attachments/files/17176702/goxlr-daemon.log)

**Comment by FrostyCoolSlug** [comment]
Could you let me know what distro you're running? It's possible it's using an old `alsa-ucm-conf` package which had a misconfigured sampler channel..

**Comment by spider312** [comment]
I'm under Ubuntu 23.04

**Comment by FrostyCoolSlug** [comment]
Could you let me know the version of `alsa-ucm-conf` that's currently installed on your system?

**Comment by spider312** [comment]
Version: 1.2.6.3-1ubuntu9.3

**Comment by FrostyCoolSlug** [comment]
Ok, that would unfortunately be why, the GoXLR Sampler configuration was fixed in 1.2.7, so you're currently running a known broken version.

I'm not really sure what to suggest outside of upgrading, but it has a dependency on libalsa as well.. The easiest way to fix it for now would be:

* Edit (as root) `/usr/share/alsa/ucm2/USB-Audio/GoXLR/GoXLR-HiFi.conf`
* Find the Line that reads: `			strings [ "goxlr_stereo_in:" $CARD ",16,17" ]`
* Change it to: `			strings [ "goxlr_stereo_in:" $CARD ",4,5" ]`

Then reboot your system, that should reconfigure the sink to use the correct channels on the GoXLR.

**Comment by spider312** [comment]
Good game ! I modified the file as you told, and it seems to work perfectly

Thank you very much, it looked like a pretty hard problem to solve :)
